### PR TITLE
Rect intersection Improve to support only border overlap

### LINKED ISSUE #15254: Rect_ intersection improve to support only border overlap detect
##### Detailed description

template Rect_ union operator will return Rect(0,0,0,0) when two Rects have only border or point overlap, I think better to keep the overlap info, meaning may return a Rect who's width or/and height may be zero when there is only border or point overlap. 

##### Steps to reproduce

Rect(0, 0, 1, 1) & Rect(1, 0, 1, 1) will got Rect(0, 0, 0, 0) instead of Rect(1, 0, 0, 1)

### PR DESCRIPTION
**Pull request by iamgd67**
resolves #15254 
### This pullrequest changes

detect the situation when two Rect has only border or point overlap, and return the overlap info by a Rect who's width or/and height may be zero

### CODE REVIEW

**Comment by iamgd67** [comment]
win64 compile build fails because of test_videoio, looks like got nothing to do with this pull request

File: modules/core/test/test_rect.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+    TEST(Rect, rectUnionOperator) {
```
Comment (dkurt) [inline modules/core/test/test_rect.cpp]: Intersection?

Comment (dkurt) [reply modules/core/test/test_rect.cpp]: I believe this test should be nearby similar ones.

Comment (iamgd67) [reply modules/core/test/test_rect.cpp]: it should be intersection, I will change it, both here and in the pr title and issue title, sorry for the language mistaken

**Comment by alalek** [comment]
Closing, see issue's comment.

**Pull request closed without merging**